ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -244,5 +294,26 @@ private KafkaStreams buildStreams(
     return new KafkaStreams(builder.build(), new StreamsConfig(newStreamsProperties));
   }
 
+  private boolean enforceKeyEquivalence(Field sinkKeyField, Field resultKeyField) {
+    if (sinkKeyField == null && resultKeyField == null) {
+      return true;
+    }
+    if (sinkKeyField != null && resultKeyField != null) {
+      if (sinkKeyField.name().equalsIgnoreCase(resultKeyField.name()) && sinkKeyField.schema() ==
```

Comment:
As far as I can tell, the Schema interface doesn't define an equals() function, so this will check reference equality. Is that what we want?

Reply:
Never mind